lib/controllers/github.js

```diff
@@ -1,42 +1,48 @@
 const { Router } = require('express');
 const { exchangeCodeForToken, getGithubProfile } = require('../utils/github');
-const authenticate = require('../middleware/authenticate');
 const GithubUser = require('../models/GithubUser');
 const jwt = require('jsonwebtoken');
 
 const ONE_DAY_IN_MS = 1000 * 60 * 60 * 24;
 
 module.exports = Router()
-  .get('/login', async (req, res) => {
+  .get('/login', (req, res) => {
     res.redirect(
       `https://github.com/login/oauth/authorize?client_id=${process.env.CLIENT_ID}&scope=user&redirect_uri=${process.env.REDIRECT_URI}`
     );
   })
 
-  .get('/login/callback', async (req, res, next) => {
-    try {
-      const { code } = req.query;
-      const token = await exchangeCodeForToken(code);
-      const { login, avatar_url, email } = await getGithubProfile(token);
-      let user = await GithubUser.findByUsername(login);
-      if (!user)
-        user = await GithubUser.insert({
-          username: login,
-          avatar: avatar_url,
-          email,
-        });
-      const payload = jwt.sign(user.toJSON(), process.env.JWT_SECRET, {
-        expiresIn: '1 day',
+  .get('/login/callback', (req, res, next) => {
+    let current = {};
+    return exchangeCodeForToken(req.query.code)
+      .then((token) => getGithubProfile(token))
+      .then((profile) => {
+        current = profile;
+        return GithubUser.findByUsername(profile.username);
+      })
+      .then((user) => {
+        if (!user)
+          return GithubUser.insert({
+            username: current.username,
+            email: current.email,
+            avatar: current.avatar_url,
+          })
```

Comment:
We should also just return the `user` coming from line 21 here too, this will ensure that if a user did exist in your database it would be passed along to the next `.then` also